core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -406,13 +449,29 @@ public JSONObject reduce(List<JSONObject> jsonList) {
         /**
          * Returns all the registered {@link Downloadable}s.
          */
+        @NonNull
         public static ExtensionList<Downloadable> all() {
             return ExtensionList.lookup(Downloadable.class);
         }
 
+        /**
+         * Returns the {@link Downloadable} that has an ID associated with the specified class.
```

Comment:
This seems to lack an explanation of the ways this can go wrong. I don't think _ID associated with the specified class_ to someone unfamiliar with internals will make sense without more context.

Also e.g. `{@see #idFor(Class)}` could help.

Reply:
Fair. I added a link to `idFor` from the JavaDoc of the relevant constructors too.